torchx/schedulers/kubernetes_mcad_scheduler.py

```diff
@@ -527,11 +527,12 @@ def app_to_resource(
     genericitems = []
 
     unique_id_size = get_unique_id_size(app.name, len(app.roles))
-    if unique_id_size == 0:
-        raise ValueError(
-            "Name size has too many characters for some Kubernetes objects. Refer to your \
-TorchX component documentation for options to change the job name."
-        )
+    if unique_id_size == 3:
```

Comment:
might be best to just make a new make_unique_trunc method that does this all rather than inline+get_unique_id_size.

That way you could test this logic directly

Also checking for size exactly equal to 3 seems a bit fragile

Reply:
Update to a consolidated function. commit: 191859d3356f8c8effac9cea2a9e2f6a39a1f728